### PR TITLE
 Add support for Encoders mapped into the keymap array

### LINKED ISSUE #9335: [Bug] Bug on split slave side using I2C comms and OLED
I'm having an issue getting my handwired split board working with I2C communication and an OLED.

There's a repeatable issue that only affects the slave, which is why I consider it a bug. Not an address conflict or config issue as far as I can tell. Something funky about split keyboard slave just not behaving over I2C if the OLED is also enabled.

How to reproduce:
If I start with OLED_DRIVER_ENABLE=OFF and flash that to the slave half, then turn it ON and flash the master half, everything works as expected, I get input from both halves and OLED works. But then if I flash the same with it ON to both halves, I stop getting input from the slave, while the OLED still works. The behavior follows whichever is the slave if I switch which side the USB is plugged in. 

Troubleshooting steps:
Sometimes I see the rgb animation slow down a ton on the master which seems to indicate an I2C timeout. Not sure how else to troubleshoot. A few posts seem to indicate this should be totally doable, but don't want to get as deep into it as something like the Helix fork to get working.

- Keyboard:  
   - Split 5x6 Dactyl variant, handwired
   - 2x Pro Micro
   - I2C between halves
   - RGB lighting + split rgb enabled
   - 1306 OLED on single half

 - AVR GCC version: 8.3.0
 - QMK Firmware version: 0.8.192

### PR DESCRIPTION
**Pull request by drashna**
## Description

This adds the ability to add positions in the LAYOUT for keyboards, to be able to use/set the encoder functionality.  This is aimed at making encoders easier to support without custom code at the keyboard/keymap level. 

To Do: 
- [x] Add Documentation
- [ ] Finalize naming conventions
- [x] Tweak setup? 
- [ ] Get working on both halves of split keyboard

## Types of Changes

- [x] Core
- [x] Enhancement/optimization
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/feature_encoders.md [hunk 1]
```diff
@@ -48,6 +48,21 @@ If you are using different pinouts for the encoders on each half of a split keyb
...
+
+## Mapping Encoder Functions to the Keymap
+
+By enabling this config, you can place the keycodes for the encoders directly in the keymap itself.  If you have extra spots in your matrix (eg, not populated), you can use those locations.  To eanble this you need to add a define that lists the row and column location to use. 
+
+```c
+#define ENCODER_KEYMAPPING { { 4, 0 }, { 4, 1 } }, { { 5, 0 }, { 5, 1 } }
+```
+
+This tells it to use 2ow 4, column 0 for counter clockwise, row 4 column 1 for clockwise, and then row 5 column 0 and 1 for the second encoder.
```
Comment (t-8ch) [inline docs/feature_encoders.md:61]: 2ow -> row

File: quantum/encoder.c [hunk 2]
```diff
@@ -185,7 +185,7 @@ void encoder_update_raw(uint8_t* slave_state) {
 #    else
 #        define NUM_ENCODERS (NUMBER_OF_ENCODERS)
 #    endif
-uint8_t    encoder_keypos[NUM_ENCODERS][2][2] = {ENCODER_KEYMAPPING};
+uint8_t    encoder_keypos[NUM_ENCODERS][2][2] = { ENCODER_KEYMAPPING};
```
Comment (t-8ch) [inline quantum/encoder.c:188]: the spacing is now inconsistent

File: docs/feature_encoders.md [hunk 3]
```diff
@@ -48,6 +48,26 @@ If you are using different pinouts for the encoders on each half of a split keyb
 #define ENCODER_RESOLUTIONS_RIGHT { 2, 4 }
 ```
 
+?> Note: When using Encoders with split keyboards, it assumes that you have double the number of encoders. If not all encoders are populated, then you may need to skip indexes. 
+
+## Mapping Encoder Functions to the Keymap
+
+By enabling this config, you can place the keycodes for the encoders directly in the keymap itself.  If you have extra spots in your matrix (eg, not populated), you can use those locations.  To eanble this you need to add a define that lists the row and column location to use.
```
Comment (t-8ch) [inline docs/feature_encoders.md:55]: eanble -> enable

Comment (drashna) [reply docs/feature_encoders.md:55]: Just a heads up, you can use the "suggestion" feature to propose code changes with the corrects.  

But thanks, I spell good. 😆  (noroads will attest).

```suggestion
By enabling this config, you can place the keycodes for the encoders directly in the keymap itself.  If you have extra spots in your matrix (eg, not populated), you can use those locations.  To enable this you need to add a define that lists the row and column location to use. 
```

File: docs/feature_encoders.md (hunk 1 above)
Comment (drashna) [reply docs/feature_encoders.md:61]:
```suggestion
This tells it to use row 4, column 0 for counter clockwise, row 4 column 1 for clockwise, and then row 5 column 0 and 1 for the second encoder.  
```

File: quantum/encoder.c (hunk 2 above)
Comment (drashna) [reply quantum/encoder.c:188]:
```suggestion
uint8_t    encoder_keypos[NUM_ENCODERS][2][2] = { ENCODER_KEYMAPPING };
```

File: docs/feature_encoders.md (hunk 3 above)
Comment (t-8ch) [reply docs/feature_encoders.md]: Will do in the future :-)

**Comment by drashna** [comment]
This mostly works.  However, it doesnt seem to respect the "slave" side encoder, without some nasty hacks

**Comment by drashna** [comment]
Closed in favor of #13286

**Pull request closed without merging**